app/components/show/barcode_component.rb

```diff
@@ -2,16 +2,20 @@
 
 module Show
   class BarcodeComponent < ApplicationComponent
-    def initialize(change_set:, version_service:)
-      @change_set = change_set
-      @version_service = version_service
+    def initialize(presenter:)
+      @presenter = presenter
     end
 
     def barcode
-      @change_set.barcode || 'Not recorded'
+      change_set.barcode || 'Not recorded'
     end
 
-    delegate :open_and_not_assembling?, to: :@version_service
-    delegate :id, to: :@change_set
+    def edit?
+      !user_version_view? && open_and_not_assembling?
```

Comment:
Here and many places below: Not showing edit pencil when a user version view.

Reply:
This branch adds nine identical definitions of `#edit?`. I wonder if this method should move to the presenter. Then `Show::SourceIdComponent#edit?` could look like 

```ruby
def edit?
  presenter.edit? && item?
end
```

❓

Reply:
In the context of a component, a method named `edit?` makes sense, as you can easily tell what is to be edited. In the context of the presenter, it isn't clear what `edit?` could apply to (edit what?) as there are many parts of the page that support editing.

Can you think of a name for the presenter method that would be clearer than `edit?`? Otherwise, I'd be inclined to leave it the way it is.

Reply:
A name like `editable?` might make the intention behind this method clearer, namely that it is intended to determine whether or not to present edit controls to the user. As for making clearer what is to be edited, I think a developer could reasonably conclude it's an object in Argo (given that the ArgoShowPresenter is used to show objects in Argo).

Reply:
Totally, but of all of the things on the show page that can update an object, which do `editable?` refer to (because it isn't all of them)?

Reply:
If it isn't all of them, then I'm fine with the repetition. A misleading abstraction is worse than repetition.